### PR TITLE
Ka/resource loader

### PR DESCRIPTION
**Pull request by kalagarraj**
Changes contain a new tool to lint and verify all commerce-sdk endpoints in a docker container. 
Refer to tools/USAGE.md for detailed information on tool's usage.

A tool like this will be helpful to verify any changes to raml-toolkit does not break existing linter rules on all commerce-sdk endpoints

FYI, this is not part of any GUS ticket, so Acceptance Criteria could be something like

1. As a salesforce steelarc developer, I am able to run this tool by following tools/USAGE.md
2. All commerce-sdk Raml definitions from commerce-sdk/apis/api-config.json should be linted in a docker container
3. Failure of one commerce-sdk raml definition MUST NOT block linting other definitions
4. Linter output should be printed in the console

### CODE REVIEW

File: tools/USAGE.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+* Git clone [raml-toolkit](https://github.com/SalesforceCommerceCloud/raml-toolkit) repository
+* Git clone [commerce-sdk](https://github.com/SalesforceCommerceCloud/commerce-sdk) repository
+
+## Usage
+
+The docker container requires `raml-toolkit` directory mounted as `/linter` volume and 
+`commerce-sdk/apis` directory mounted as `/apis` volume. You can do so as follows
+
+```shell script
+$ docker run -t -d --name linter -v /<<raml-toolkit absolute path>>:/linter -v /<<commerce-sdk/apis absolute path>>:/apis node
```
Comment (bredmond-sf) [inline tools/USAGE.md:24]: Best practice is to use a tag for the image to make sure this stays stable. I'd recommend 12-alpine
Also, you should be able to just add sh /linter/tools/lint.sh onto the end of this and skip the second command

Comment (bredmond-sf) [reply tools/USAGE.md:24]: do you mean a docker-compose file? It could just be another shell script

Comment (unandyala) [reply tools/USAGE.md:24]: not docker-compose, just the docker file that has instructions to run the linter inside docker instead of mounting

Comment (bredmond-sf) [reply tools/USAGE.md:24]: I think you mean just having another shell script. There's not a very good reason to build a custom docker image.

Comment (kalagarraj) [reply tools/USAGE.md]: Docker file is a little heavy for this one.. In the latest change, i even deleted the shell script

Pls take a look

Comment (bredmond-sf) [reply tools/USAGE.md]: Yup, looks very good. One other thing to consider is adding a note about storing the command as a bash alias.
```
alias linter-test="docker run -t --name linter -v /<<raml-toolkit absolute path>>:/linter -v /<<commerce-sdk/apis absolute path>>:/apis node:12-alpine /linter/tools/lint.js"
linter-test /linter/commerce-apps-raml-toolkit-0.3.1.tgz
```

File: tools/USAGE.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+```shell script
+$ docker run -t --name linter -v /<<raml-toolkit absolute path>>:/linter -v /<<commerce-sdk/apis absolute path>>:/apis node:12-alpine /linter/tools/lint.js
+```
+
+The tool uses latest `raml-toolkit` npm library by default to check the endpoints. 
+The default can be overridden by providing another `raml-toolkit` npm library as a parameter.
+
+```shell script
+$ docker run -t --name linter -v /<<raml-toolkit absolute path>>:/linter -v /<<commerce-sdk/apis absolute path>>:/apis node:12-alpine /linter/tools/lint.js @commerce-apps/raml-toolkit@0.2.9
```
Comment (unandyala) [inline tools/USAGE.md:31]: nice

File: tools/USAGE.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  
+You can also build `raml-toolkit` npm library as follows from your local `raml-toolkit` folder
+
+```nashorn js
+$ npm pack
+```
+The `npm pack` command creates a file similar to `commerce-apps-raml-toolkit-x.x.x.tgz`. Providing this file as follows overrides the default npm library that checks the endpoints. 
+
+```shell script
+$ docker run -t --name linter -v /<<raml-toolkit absolute path>>:/linter -v /<<commerce-sdk/apis absolute path>>:/apis node:12-alpine /linter/tools/lint.js /linter/commerce-apps-raml-toolkit-0.3.1.tgz
```
Comment (unandyala) [inline tools/USAGE.md:42]: nice.. this is very useful

**Pull request merged**